File: test-data/typescript/design_patterns/singleton.ts
export class Singleton {
    private static instance: Singleton;
    public value: string;

    private constructor() {
        this.value = "I am the only one";
    }

    public static getInstance(): Singleton {
        if (!Singleton.instance) {
            Singleton.instance = new Singleton();
        }
        return Singleton.instance;
    }
}

File: test-data/java/design_patterns/Singleton.java
package com.example.patterns;

public class Singleton {
    private static Singleton instance;
    public String value;

    private Singleton() {
        this.value = "I am the only one";
    }

    public static Singleton getInstance() {
        if (instance == null) {
            instance = new Singleton();
        }
        return instance;
    }
}
